src/api/Feed.js

```diff
@@ -1178,6 +1180,58 @@ class Feed extends Base {
         return pendingFeedItem;
     };
 
+    /**
+     * Add a placeholder pending comment (reply).
+     *
+     * @param {string} parentId - id of parent comment or annotation
+     * @param {Object} currentUser - the user who performed the action
+     * @param {Object} commentBase - Base properties for reply (comment) to be added to the feed as pending.
+     * @return {Comment} - newly created pending reply
+     */
+    addPendingReply = (parentId: string, currentUser: User, commentBase: Object): Comment => {
+        if (!this.file.id) {
+            throw getBadItemError();
+        }
+        if (!currentUser) {
+            throw getBadUserError();
+        }
+
+        const date = new Date().toISOString();
+        const pendingReply: Comment = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...commentBase,
+        };
+
+        const getItemWithPendingReply = <T: { replies?: Array<Comment> }>(item: T, reply: Comment): T => {
+            const updatedItem: T = { ...item };
+            if (!updatedItem.replies) {
+                updatedItem.replies = [];
+            }
+            updatedItem.replies.push(reply);
+            return updatedItem;
+        };
+
+        const cachedItems = this.getCachedItems(this.file.id);
+        if (cachedItems) {
+            const updatedFeedItems = cachedItems.items.map(item => {
+                if (item.id === parentId && item.type === 'comment') {
+                    return getItemWithPendingReply<Comment>(item, pendingReply);
+                }
+                if (item.id === parentId && item.type === 'annotation') {
+                    return getItemWithPendingReply<Annotation>(item, pendingReply);
+                }
+                return item;
+            });
```

Comment:
Do we need the typed helper method? Could it be something like
```
cachedItems.items.map(item => {
    if (item.id !== parentId) {
        return item;
    }

    return {
        ...item,
        replies: item.replies ? [...item.replies, pendingReply] : [pendingReply],
    }
})
```

Similarly, for the `updateReplyItem` method